File: ios/App/App/OmMetaPlugin.m
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(OmMetaPlugin, "OmMeta",
    CAP_PLUGIN_METHOD(log, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(requestReview, CAPPluginReturnPromise);
)

File: ios/App/App/OmMetaPlugin.swift
import Capacitor
import FBSDKCoreKit
import Foundation
import StoreKit

@objc(OmMetaPlugin)
public class OmMetaPlugin: CAPPlugin {
    @objc func log(_ call: CAPPluginCall) {
        let value = call.getString("event") ?? ""
        AppEvents.shared.logEvent(AppEvents.Name(value))
        call.resolve(["success": true])
    }

    @objc func requestReview(_ call: CAPPluginCall) {
        SKStoreReviewController.requestReview()
        call.resolve()
    }
}

